src/app/App.js

```diff
@@ -510,10 +518,18 @@ class App extends EventEmitter {
 	 * @return {string}
 	 */
 	getRoutePath(path) {
-		path = utils.getUrlPathWithoutHash(path);
-		// Makes sure that the path substring will be in the expected format
-		// (that is, will end with a "/"), even after removing the base path.
-		return utils.getUrlPathWithoutHash(path.substr(this.basePath.length));
+		if (this.ignoringQueryString) {
```

Comment:
Yeah. I think it should be fine. It could be something like:

"Returns the given path formatted to be matched by a route. This will, for example, remove the base path from it, but make sure it will end with a '/'"